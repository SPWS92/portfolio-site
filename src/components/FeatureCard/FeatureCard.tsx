const FeatureCard = ({ children: icon, feature }) => {
  return (
    <div className='w-full h-full border rounded-lg p-8 shadow-xl'>
      {icon}
      <p className='text-xl font-bold'>{feature.heading}</p>
      {feature?.description ?? <p className='text-lg text-gray-700'>{feature.description}</p>}
    </div>
  )
}

export default FeatureCard;