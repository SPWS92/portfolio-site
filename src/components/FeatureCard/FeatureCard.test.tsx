import { render } from '@testing-library/react'

import FeatureCard from './FeatureCard.tsx';

const featureStub = {
  id: 1,
  heading: 'Mock Heading',
  description: 'This is a mock description'
};

describe('Components: FeatureCard', () => {
  it('SHOULD render as expected', () => {
    const component = render(
      <FeatureCard feature={featureStub}>
        <p>Mock child</p>
      </FeatureCard>
    );

    expect(component).toMatchSnapshot();
  });

  it('SHOULD no render description if description does not exist', () => {
    const component = render(
      <FeatureCard feature={{ id: 1, heading: 'Mock Heading' }}>
        <p>Mock child</p>
      </FeatureCard>
    );

    expect(component).toMatchSnapshot();
  });
})
