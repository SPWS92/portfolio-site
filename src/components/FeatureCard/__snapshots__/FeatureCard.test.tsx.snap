// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: FeatureCard SHOULD no render description if description does not exist 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-full border rounded-lg p-8 shadow-xl"
      >
        <p>
          Mock child
        </p>
        <p
          class="text-xl font-bold"
        >
          Mock Heading
        </p>
        <p
          class="text-lg text-gray-700"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-full border rounded-lg p-8 shadow-xl"
    >
      <p>
        Mock child
      </p>
      <p
        class="text-xl font-bold"
      >
        Mock Heading
      </p>
      <p
        class="text-lg text-gray-700"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components: FeatureCard SHOULD render as expected 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-full border rounded-lg p-8 shadow-xl"
      >
        <p>
          Mock child
        </p>
        <p
          class="text-xl font-bold"
        >
          Mock Heading
        </p>
        This is a mock description
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-full border rounded-lg p-8 shadow-xl"
    >
      <p>
        Mock child
      </p>
      <p
        class="text-xl font-bold"
      >
        Mock Heading
      </p>
      This is a mock description
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
